Buddies.Web/src/stores/authStore.ts

```diff
@@ -0,0 +1,20 @@
+﻿import create from 'zustand';
+import { JwtPayload } from 'jwt-decode';
+
+// look into generating this from the backend in the future
+export interface AuthState extends JwtPayload {
+  /**
+   * ID of the currently logged in user.
+   */
+  nameid: string,
+  /**
+   * First name of the currently logged in user.
+   */
+  given_name: string,
+  /**
+   * Last name of the currently logged in user.
+   */
+  family_name: string
```

Comment:
id, firstName, lastName?

Reply:
By default, ASP.NET Core enforces the key names used inside token JWT payloads through the `ClaimTypes` constants ([example](https://github.com/GDSCUTM-CommunityProjects/Buddies/blob/4fc53c726b017f373c0497b25472ed5bc2f3b866/Buddies.API/Services/TokenService.cs#L50)). We can look into overriding these in the future (similar to [here](https://github.com/GDSCUTM-CommunityProjects/Buddies/blob/feat/user-login/Buddies.API/IO/CustomIdentityErrorDescriber.cs)) but we are stuck with the names from `ClaimTypes` for now.